pkg/commands/apply/apply.go

```diff
@@ -319,6 +327,32 @@ func (c *ApplyCommand) Process(ctx context.Context) (merr error) {
 	return merr
 }
 
+func (c *ApplyCommand) resolveGitHubLogURL(ctx context.Context) string {
+	logger := logging.FromContext(ctx)
+
+	// Default to action summary page
+	logURL := fmt.Sprintf("[[logs](%s/%s/%s/actions/runs/%d/attempts/%d)]", c.cfg.ServerURL, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.cfg.RunID, c.cfg.RunAttempt)
+
+	if !c.FlagIsGitHubActions {
+		logger.DebugContext(ctx, "skipping github log url resolution", "is_github_action", c.FlagIsGitHubActions)
+		return logURL
+	}
+
+	// Link to specific job's logs directly if possible
+	jobs, err := c.gitHubClient.ListJobsForWorkflowRun(ctx, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.cfg.RunID, nil)
+	if err != nil {
+		logger.DebugContext(ctx, "failed to list jobs for workflow run", "err", err)
+	} else {
+		for _, job := range jobs.Jobs {
+			if c.flagJobName == job.Name {
+				logURL = fmt.Sprintf("[[logs](%s/%s/%s/actions/runs/%d/job/%d)]", c.cfg.ServerURL, c.GitHubFlags.FlagGitHubOwner, c.GitHubFlags.FlagGitHubRepo, c.cfg.RunID, job.ID)
```

Comment:
Maybe change this to a `return` to simplify a bit? Or do we specifically need the last job with the given name (in which case please add a code comment explaining how/why there are multiple matching jobs)?

Reply:
Refactored this whole bit of logic into the github client and used early return there.